src/core/thread/mle_router.cpp

```diff
@@ -2035,6 +2035,7 @@ ThreadError MleRouter::HandleChildUpdateRequest(const Message &aMessage, const I
     }
 
     child->mLastHeard = Timer::GetNow();
+    child->mAddSrcMatchEntryShort = true;
```

Comment:
This code will be kept here to handle situations that Child Update Request message which uses extended address arrives before other messages with short address.
But will update the code as your suggestion to handle the situation that Child Update Request message is not immediately sent after attaching.